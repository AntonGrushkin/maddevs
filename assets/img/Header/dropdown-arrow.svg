<svg width="9" height="6" viewBox="0 0 9 6" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M0.375 0.4375L4.49992 4.5625L8.5 0.4375" stroke="#828282"/>
</svg>
