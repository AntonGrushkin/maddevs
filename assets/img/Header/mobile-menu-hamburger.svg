<svg width="28" height="14" viewBox="0 0 28 14" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M1 7H27M1 1H27M1 13H27" stroke="#F2F2F2" stroke-linecap="round" stroke-linejoin="round"/>
</svg>
