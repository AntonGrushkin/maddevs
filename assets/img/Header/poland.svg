<svg width="18" height="15" viewBox="0 0 18 15" fill="none" xmlns="http://www.w3.org/2000/svg">
<g clip-path="url(#clip0)">
<path fill-rule="evenodd" clip-rule="evenodd" d="M18 14.5H0V0.5H18V14.5Z" fill="white"/>
<path fill-rule="evenodd" clip-rule="evenodd" d="M18 14.5H0V7.5H18V14.5Z" fill="#DC143C"/>
</g>
<defs>
<clipPath id="clip0">
<rect y="0.5" width="18" height="14" fill="white"/>
</clipPath>
</defs>
</svg>
