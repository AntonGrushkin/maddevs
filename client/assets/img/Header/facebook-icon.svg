<svg width="23" height="23" viewBox="0 0 23 23" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M3.813 22.5H18.937C19.8488 22.5 20.7233 22.1378 21.368 21.493C22.0128 20.8483 22.375 19.9738 22.375 19.062V3.938C22.375 3.02619 22.0128 2.15172 21.368 1.50697C20.7233 0.862217 19.8488 0.5 18.937 0.5H3.813C3.36148 0.499869 2.91436 0.588706 2.49718 0.761435C2.08 0.934164 1.70095 1.1874 1.38167 1.50667C1.0624 1.82595 0.809164 2.205 0.636435 2.62218C0.463706 3.03936 0.374869 3.48648 0.375 3.938V19.062C0.375 19.9738 0.737217 20.8483 1.38197 21.493C2.02672 22.1378 2.90119 22.5 3.813 22.5ZM8.723 9.667H9.859V8.53C9.80682 7.93767 9.93474 7.3434 10.226 6.825C10.4265 6.55077 10.6928 6.33145 11.0004 6.18727C11.308 6.04309 11.647 5.97868 11.986 6C12.6725 5.9689 13.3599 6.03693 14.027 6.202L13.74 7.888C13.4409 7.80646 13.133 7.76146 12.823 7.754C12.383 7.754 11.986 7.913 11.986 8.365V9.667H13.819L13.691 11.311H11.986V17H9.859V11.286H8.723V9.666V9.667Z" fill="#96969C"/>
</svg>
