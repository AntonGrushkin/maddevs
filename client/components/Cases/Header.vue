<template>
  <section class="case_header">
    <div class="case_logotype nambafood"></div>
    <h1 class="case_title">Namba Food: <br> Top 1 Delivery Service <br> in Central Asia</h1>
    <button type="button" class="case_more__button">Read our case studies â†“</button>
    <video class="case_video" autoplay="true" loop="true" muted="muted">
      <source src="../../assets/img/Cases/nambafood/video/nambafood-header-video.mp4" type="video/mp4">
    </video>
  </section>
</template>

<script>
export default {
  name: 'Header'
};
</script>

<style scoped lang="scss">
@import "../../assets/styles/cases/header";
@import "../../assets/styles/cases/mixins";

.case {
  &_header {
    height: 100vh;
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
    position: relative;
    background-color: transparent;
  }

  &_logotype {
    @include nambafood-header;
    width: 127px;
    height: 114px;
    margin-bottom: 54px;
  }

  &_title {
    @include title($text-color--white, 51.88px, -2px);
    text-align: center;
  }

  &_more__button {
    position: absolute;
    bottom: 0;
    max-width: 1104px;
    height: 55px;
    width: 100%;
    background-color: $bgcolor--pink;
    border: 0;
    cursor: pointer;

    @include default_text($text-color--white, 18px, 30px, -0.035em, normal);
  }

  &_video {
    width: auto;
    height: 100%;
    position: absolute;
    background-color: $bgcolor--red;
    z-index: -1;
  }
}
</style>
