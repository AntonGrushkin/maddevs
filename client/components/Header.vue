<template>
  <div class="header-wrapper">
    <header ref="header" class="header">
      <div class="container">
        <div class="header__header-content_wrap">
          <div class="header__left-nav_bar">
            <router-link :to="`/`" class="header__logo-icon">
              <headerLogo class="header__header-logo" :headerLogoTextDisplayState="headerLogoTextDisplayState"/>
            </router-link>
            <nav class="header__header-routes_links">
              <router-link @click.native="goToTopPage" exact to="/">About</router-link>
              <router-link @click.native="goToTopPage" to="/services">Services</router-link>
              <router-link @click.native="goToTopPage" to="/projects">Projects</router-link>
              <router-link @click.native="goToTopPage" to="/careers">Careers</router-link>
              <a href="https://blog.maddevs.io/" target="_blank" rel="noreferrer">Blog</a>
            </nav>
          </div>
          <div class="header__right-content">
            <div class="header__right-text_content">
              <div class="header__soc-links_wrap">
                <a href="https://twitter.com/maddevsio" target="_blank" class="header__twitter-link header__soc-link">
                  <img src="@/assets/img/Header/twitter-icon.svg" alt="Twitter">
                </a>
                <a href="https://ru.linkedin.com/company/mad-devs" target="_blank" class="header__lindekin-link header__soc-link">
                  <img src="@/assets/img/Header/lindekin-icon.svg" alt="Lindekin">
                </a>
                <a href="https://www.facebook.com/maddevsllc" target="_blank" class="header__facebook-link header__soc-link">
                  <img src="@/assets/img/Header/facebook-icon.svg" alt="Facebook">
                </a>
              </div>
              <div class="header__phones-dropdown_wrap" :id="selectedPhone.country">
                <a class="header__selected-phone" :href="`tel:${selectedPhone.phoneNumber}`">
                  <img :src="require(`@/assets/img/Flags/${selectedPhone.country}.svg`)" :alt="`${selectedPhone.country}`" />
                  {{ selectedPhone.phoneNumber }}
                </a>
                <div class="header__phones-list">
                  <a v-for="(phone, i) in phones" :key="i" class="header__phone-item" :href="`tel:${phone.phoneNumber}`" @click="selectedPhone = phone" v-show="phone != selectedPhone">
                    <img :src="require(`@/assets/img/Flags/${phone.country}.svg`)" :alt="`${phone.country}`" />
                    {{ phone.phoneNumber }}
                  </a>
                </div>
              </div>
              <a href="mailto:team@maddevs.io" class="header__mailto-link">team@maddevs.io</a>
            </div>
            <UIButtonModalTrigger 
              :buttonInnerText="buttonInnerText" 
              :modalWindowName="modalWindowName" 
              :isRed="true"
            />
          </div>
        </div>
      </div>
    </header>
    <mobileHeader />
  </div>
</template>

<script>
import UIButtonModalTrigger from '@/components/ui/UIButtonModalTrigger';
import mobileHeader from '@/components/ui/mobile-header';
import headerLogo from '@/components/svg/headerLogo';

export default {
  name: 'main-header',
  components: {
    UIButtonModalTrigger,
    mobileHeader,
    headerLogo
  },
  data() {
    return {
      buttonInnerText: 'Contact me',
      phones: [
        // {
        //   phoneNumber: '+1 833-MADDEVS',
        //   country: 'united-states'
        // },
        {
          phoneNumber: '+44 20 3984 8555',
          country: 'united-kingdom'
        }
      ],
      selectedPhone: null,
      modalWindowName: 'contact-me-modal',
      headerLogoTextDisplayState: false
    };
  },
  created() {
    /**
     * Sets default value for selectedPhone
     * this.phones[1] is a United Kingdom number
    */
    this.selectedPhone = this.phones[0];
  },
  watch: {
    '$route'() {
      this.headerLogoTextDisplayState = false;
    }
  },
  mounted() {
    window.addEventListener('scroll', this.handleScroll);
    this.handleScroll();
  },
  destroyed() {
    window.removeEventListener('scroll', this.handleScroll);
  },
  methods: {
    goToTopPage() {
      window.scrollTo(0, 0);
    },
    handleScroll() {
      if(window.pageYOffset >= 100)
        this.headerLogoTextDisplayState = true;
      else 
        this.headerLogoTextDisplayState = false;
    }
  }
};
</script>

<style lang="scss" scoped>
  @import '../assets/styles/vars';

  .header {
    width: 100%;
    height: 40px;
    padding: 11px 0;
    position: fixed;
    z-index: 2;
    background-color: #000;

    button {
      width: 135px;
      height: 38px;
      margin-right: -96px;
    }

    &__header-logo {
      width: 40px;
      height: 70px;
    }

    &__header-content_wrap {
      height: max-content;
      display: flex;
      justify-content: space-between;
    }

    &__header-routes_links {
      position: relative;
      padding-top: 9px;
      padding-left: 55px;

      a {
        color: $text-color--grey;
        text-decoration: none;
        margin-right: 16px;
        font-size: 17px;
        font-family: 'Hoves-Regular';

        &::after {
          content: 'â†“';
          color: transparent;
        }

        &:last-child {
          &::after {
            content: '';
          }
        }
      }

      .nuxt-link-active {
        color: $text-color--red;

        &::after {
          color: $text-color--red;
        }
      }
    }

    &__left-nav_bar,
    &__right-content,
    &__right-text_content {
      display: flex;
    }

    &__right-text_content {
      display: flex;
      padding-top: 9px;
    }

    &__soc-links_wrap {
      padding-right: 26px;
    }

    &__soc-link {
      padding-right: 5px;
    }

    &__phones-dropdown_wrap {
      // min-width: 200px;
      padding-top: 1px;
      // background: url('../assets/img/Header/dropdown-arrow.svg') no-repeat;
      background-position-y: 9px;
      background-position-x: 170px;

      &:hover {
        .header__phones-list {
          display: flex;
          flex-direction: column;
        }
      }
    }

    &__phones-list {
      display: none;
      position: absolute;
      margin-left: -10px;
      padding: 0 10px 10px;
      background-color: #000;
    }

    &__mailto-link,
    &__selected-phone,
    &__phone-item {
      padding-right: 26px;
      color: $text-color--grey;
      font-size: 17px;
      font-family: 'Hoves-Regular';
      text-decoration: none;
    }

    &__selected-phone,
    &__phone-item {
      position: relative;
      padding: 0 25px;
      cursor: pointer;

      img {
        position: absolute;
        left: 0;
      }
    }

    &__selected-phone {
      img {
        top: 4px;
      }
    }

    &__phone-item {
      padding-top: 17px;

      img {
        top: 20px;
      }
    }

    &__logo-icon {
      margin-left: -96px;
    }
  }

  .mobile-menu_is-open {
    width: 100%;
    height: 100%;
    position: fixed;
    padding: initial;
    z-index: 999;

    .container {
      height: 100vh;
      overflow: scroll;
    }
  }

  @media only screen and (max-width: 1590px) {
    .header {
      &__logo-icon {
        margin-left: -80px;
      }

      button {
        margin-right: -80px;
      }

      &__header-routes_links {
        padding-left: 33px;
      }
    }
  }

  @media only screen and (max-width: 1320px) {
    .header {
      &__header-routes_links {
        padding-left: 25px;
      }

      &__mailto-link,
      &__selected-phone,
      &__phone-item,
      &__soc-links_wrap {
        padding-right: 15px;
      }

      &__routes_links {
        a {
          margin-right: 0;
        }
      }

      &__logo-icon {
        margin-left: -68px;
      }

      button {
        margin-right: -68px;
      }
    }
  }

  @media only screen and (max-width: 1240px) {
    .header {
      &__header-routes_links {
        padding-top: 10px;

        a {
          margin-right: 5px;
        }
      }

       &__selected-phone {
        img {
          top: 2px;
        }
      }

      &__phone-item {
        img {
          top: 19px;
        }
      }
    }
  }

  @media only screen and (max-width: 1140px) {
    .header {
      &__logo-icon {
        margin-left: -80px;
      }

      button {
        margin-right: -80px;
      }
    }
  }

  @media only screen and (max-width: 1120px) {
    .header {
      &__mailto-link,
      &__selected-phone,
      &__phone-item,
      &__soc-links_wrap {
        padding-right: 5px;
      }

      &__header-routes_links {
        a {
          margin-right: 0;
        }
      }

      &__mailto-link {
        padding-top: 1px;
      }
    }
  }

  @media only screen and (max-width: 1024px) {
    .header {
      display: none;
    }
  }
</style>
