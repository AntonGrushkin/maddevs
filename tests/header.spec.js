import {
  mount
} from '@vue/test-utils';
import Header from '@/components/Header';

describe('Header', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(Header, {
      stubs: ['router-link', 'modal', 'perfect-scrollbar']
    });
  });

  // ------ IMPORTANT ----- //
  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
  });
  
  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
  // --------------------- //

  it('has a functions', () => {
    expect(typeof wrapper.vm.mainPageHandleScroll).toBe('function');
    expect(typeof wrapper.vm.servicesPageHandleScroll).toBe('function');
    expect(typeof wrapper.vm.projectsPageHandleScroll).toBe('function');
  });

  it('sets the correct default data', () => {
    expect(typeof Header.data).toBe('function');
    const defaultData = Header.data();
    expect(defaultData.modalWindowName).toBe('contact-me-modal');
    expect(defaultData.headerIsWhite).toBe(false);
  });

  it('call function mainPageHandleScroll with different params for: getBoundingClientRect()', () => {
    global.$nuxt = {
      $route: {
        name: 'index'
      }
    };
    document.body.innerHTML =
    `<div>
      <div id="customer-testimonials"></div>
      <div id="meet-our-experts"></div>
      <div id="why-us"></div>
     </div>`;

    // if condition in functions correctly 
    Element.prototype.getBoundingClientRect = jest.fn(() => Object({top: 0, bottom: 0}));
    wrapper.vm.mainPageHandleScroll();
    expect(wrapper.vm.$data.headerIsWhite).toBe(true);

    // if condition in functions not correctly
    Element.prototype.getBoundingClientRect = jest.fn(() => Object({top: 12, bottom: 34}));
    wrapper.vm.mainPageHandleScroll();
    expect(wrapper.vm.$data.headerIsWhite).toBe(false);
  });

  it('call function servicesPageHandleScroll with different params for: getBoundingClientRect()', () => {
    global.$nuxt = {
      $route: {
        name: 'services'
      }
    };
    document.body.innerHTML =
    `<div>
      <div id="infrastructure-optimisation"></div>
      <div id="it-consulting"></div>
     </div>`;

    // if condition in functions correctly 
    Element.prototype.getBoundingClientRect = jest.fn(() => Object({top: 0, bottom: 0}));
    wrapper.vm.servicesPageHandleScroll();
    expect(wrapper.vm.$data.headerIsWhite).toBe(true);

    // if condition in functions not correctly
    Element.prototype.getBoundingClientRect = jest.fn(() => Object({top: 12, bottom: 34}));
    wrapper.vm.servicesPageHandleScroll();
    expect(wrapper.vm.$data.headerIsWhite).toBe(false);
  });

  it('call function projectsPageHandleScroll with different params for: getBoundingClientRect()', () => {
    global.$nuxt = {
      $route: {
        name: 'projects'
      }
    };
    document.body.innerHTML = '<div id="open-source"></div>';

    // if condition in functions correctly 
    Element.prototype.getBoundingClientRect = jest.fn(() => Object({top: 0, bottom: 0}));
    wrapper.vm.projectsPageHandleScroll();
    expect(wrapper.vm.$data.headerIsWhite).toBe(true);

    // if condition in functions not correctly
    Element.prototype.getBoundingClientRect = jest.fn(() => Object({top: 12, bottom: 34}));
    wrapper.vm.projectsPageHandleScroll();
    expect(wrapper.vm.$data.headerIsWhite).toBe(false);
  });

  it('headerIsWhite should equal false after call function', () => {
    wrapper.vm.$data.headerIsWhite = true;
    wrapper.vm.listenCareersLinkClick();
    expect(wrapper.vm.$data.headerIsWhite).toBe(false);
  });
});
