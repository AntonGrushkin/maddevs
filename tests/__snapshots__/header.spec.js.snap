// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div
  class="header-wrapper"
>
  <header
    class="header"
  >
    <div
      class="container"
    >
      <div
        class="header__header-content_logo"
      >
        <router-link-stub
          class="header__logo-icon"
          to="/"
        >
          <svg
            class="header__header-logo"
            viewBox="0 0 256 446"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <transition-stub
                name="fade"
              >
                <path
                  d="M205.78 270.83h-22.2a5.4 5.4 0 0 0-5.4 5.4v79.19a5.4 5.4 0 0 0 5.4 5.4h22.2c33.11 0 49.79-18 49.79-44.81 0-26.81-16.68-45.18-49.79-45.18zm7 62.6H207a3.6 3.6 0 0 1-3.6-3.6v-28.4a3.6 3.6 0 0 1 3.6-3.6h5.4c7.59 0 12.6 6.35 12.6 18.18s-4.58 17.43-12.17 17.43l-.05-.01zM5.4 360.82h16.2a5.4 5.4 0 0 0 5.4-5.4v-25.2L35.12 347a5.4 5.4 0 0 0 4.88 3h.22a5.4 5.4 0 0 0 4.78-2.83l9-16.94v25.2a5.4 5.4 0 0 0 5.4 5.4h16.2a5.4 5.4 0 0 0 5.4-5.4v-79.2a5.4 5.4 0 0 0-5.4-5.4H58.09a5.4 5.4 0 0 0-4.77 2.87L39.6 299.63l-13.73-25.92a5.4 5.4 0 0 0-4.77-2.87H5.4a5.4 5.4 0 0 0-5.4 5.4v79.19a5.4 5.4 0 0 0 5.4 5.39zM144 352.14l1.36 4.77a5.4 5.4 0 0 0 5.19 3.92h15a5.39 5.39 0 0 0 5.17-6.94c-5.39-17.9-19.62-65.18-23.76-79a5.38 5.38 0 0 0-5.16-3.83h-24.4a5.4 5.4 0 0 0-5.17 3.84l-23.75 79a5.4 5.4 0 0 0 5.17 7h15.66a5.4 5.4 0 0 0 5.19-3.92l1.36-4.77a5.4 5.4 0 0 1 5.19-3.92h17.76a5.4 5.4 0 0 1 5.19 3.85zm-21.62-24.76l7.2-25.2 7.2 25.2h-14.4zM27 377H5.4a5.4 5.4 0 0 0-5.4 5.4V440a5.4 5.4 0 0 0 5.4 5.4h23.4c24.78 0 36-14.17 36-34.2S51.77 377 27 377zm1.8 46.79h-3.6a1.8 1.8 0 0 1-1.8-1.8v-21.6a1.8 1.8 0 0 1 1.8-1.8h3.6c7 0 10.8 3.75 10.8 12.6s-3.82 12.63-10.8 12.63v-.03zM244.48 407.33c-2.32-1.553-7.293-3.3-14.92-5.24l-3.77-.93c-.267-.06-.653-.163-1.16-.31-5.74-1.447-8.61-3.253-8.61-5.42a3.79 3.79 0 0 1 1.64-3.37 9.06 9.06 0 0 1 5-1.11 10.32 10.32 0 0 1 5.88 1.42c1.38.95 3.83 2.53 4.15 4.46h13.75a3.59 3.59 0 0 0 3.51-4.41 18.63 18.63 0 0 0-7.09-10.94 33.18 33.18 0 0 0-19.86-6.26 32.71 32.71 0 0 0-20.12 6.49c-4.89 3.7-8 8.09-8 14.57a22.39 22.39 0 0 0 2.52 9.32 17.69 17.69 0 0 0 5.66 6.48 29.14 29.14 0 0 0 6.24 2.86c2.86 1.02 6.7 2.153 11.52 3.4l2.71.67c5.887 1.48 8.833 3.403 8.84 5.77a3.89 3.89 0 0 1-2.11 3.49 11.25 11.25 0 0 1-5.79 1.27c-2.933 0-5.14-.517-6.62-1.55-1.48-1-2.61-3.59-3.17-6h-15.6a3.61 3.61 0 0 0-3.54 4.33c1.09 5.26 3.39 9.84 7.4 12.9 5.08 3.88 11.38 6.16 20.05 6.16 9.41 0 17.21-2.28 22.26-6.16 5.05-3.88 7.25-10 7.25-17.24 0-3.35-.35-5.66-1.68-8.12a18.8 18.8 0 0 0-6.34-6.53zM185.94 377h-15.31a5.4 5.4 0 0 0-5.31 4.43l-6.39 35.16-6.39-35.16a5.4 5.4 0 0 0-5.31-4.43h-15.32a5.4 5.4 0 0 0-5.19 6.9l16.67 57.59a5.4 5.4 0 0 0 5.19 3.9h20.68a5.4 5.4 0 0 0 5.19-3.9l16.67-57.59a5.4 5.4 0 0 0-5.18-6.9zM115.19 377h-39.6a5.4 5.4 0 0 0-5.4 5.4V440a5.4 5.4 0 0 0 5.4 5.4h39.6a5.4 5.4 0 0 0 5.4-5.4v-9a3.6 3.6 0 0 0-3.6-3.6h-23.4a1.8 1.8 0 0 1-1.8-1.8V422a1.8 1.8 0 0 1 1.8-1.8h18a3.6 3.6 0 0 0 3.6-3.6v-10.8a3.6 3.6 0 0 0-3.6-3.6h-18a1.8 1.8 0 0 1-1.8-1.8v-3.6a1.8 1.8 0 0 1 1.8-1.8H117a3.6 3.6 0 0 0 3.6-3.6v-9a5.4 5.4 0 0 0-5.41-5.4z"
                  fill="#fff"
                  fill-rule="nonzero"
                />
              </transition-stub>
               
              <path
                d="M237.23 1.34l-109.48 50-109.48-50C10.59-2.83 0 3.16 0 13.87v208.77c0 8 4.79 14.52 11.29 14.72a3.6 3.6 0 0 0 .9 0h.65l71.75-7.45L71 210.54c-5.94-8.29 1.26-15.49 8.88-12.69 1.53.56 3.19-.18 4.68.21l-18-60.72c-.07-.21-.13-.42-.19-.63l-.14-.46v-.14a10.39 10.39 0 0 1 19.47-6.7v.09c.17.34 20.65 50.87 20.65 50.87a10.95 10.95 0 0 1 21.62.57 11 11 0 0 1 11.03-9.52c5.35 0 9.5 4.73 10.43 9.83l12.17-34.3a10.44 10.44 0 0 1 20.24 3.52v.32L171 229.84l71.67 7.45h.65a3.6 3.6 0 0 0 .9 0c6.5-.2 10.93-6.68 10.93-14.69V13.82c-.01-10.7-10.24-16.65-17.92-12.48z"
                fill="#CA2228"
              />
            </g>
          </svg>
        </router-link-stub>
         
        <div
          class="header__header-content_wrap"
        >
          <div
            class="header__left-nav_bar"
          >
            <nav
              class="header__header-routes_links"
            >
              <router-link-stub
                exact=""
                to="/"
              >
                About
              </router-link-stub>
               
              <router-link-stub
                to="/services"
              >
                Services
              </router-link-stub>
               
              <router-link-stub
                to="/projects"
              >
                Projects
              </router-link-stub>
               
              <router-link-stub
                to="/careers"
              >
                Careers
              </router-link-stub>
               
              <a
                href="https://blog.maddevs.io/"
                rel="noreferrer"
                target="_blank"
              >
                Blog
              </a>
            </nav>
          </div>
           
          <div
            class="header__right-content"
          >
            <div
              class="header__right-text_content"
            >
              <div
                class="header__soc-links_wrap"
              >
                <a
                  class="header-social-logo header-social-logo--twitter header__soc-link"
                  href="https://twitter.com/maddevsio"
                  target="_blank"
                />
                 
                <a
                  class="header-social-logo header-social-logo--linkedin header__soc-link"
                  href="https://ru.linkedin.com/company/mad-devs"
                  target="_blank"
                />
                 
                <a
                  class="header-social-logo header-social-logo--facebook header__soc-link"
                  href="https://www.facebook.com/maddevsllc"
                  target="_blank"
                />
              </div>
               
              <div
                class="header__phones-dropdown_wrap"
                id="united-states"
              >
                <a
                  class="header__selected-phone"
                  href="tel:+1 833-MADDEVS"
                >
                  <span
                    class="header__flag header__flag--united-states"
                  />
                  
                +1 833-MADDEVS
                
                  <i
                    class="header__phones-dropdown_arrow"
                  />
                </a>
                 
                <div
                  class="header__phones-list"
                >
                  <a
                    class="header__phone-item"
                    href="tel:+44 20 3984 8555"
                  >
                    <span
                      class="header__flag header__flag--united-kingdom"
                    />
                    
                  +44 20 3984 8555
                
                  </a>
                  <a
                    class="header__phone-item"
                    href="tel:+1 833-MADDEVS"
                    style="display: none;"
                  >
                    <span
                      class="header__flag header__flag--united-states"
                    />
                    
                  +1 833-MADDEVS
                
                  </a>
                </div>
              </div>
               
              <a
                class="header__mailto-link"
                href="mailto:team@maddevs.io"
              >
                team@maddevs.io
              </a>
            </div>
          </div>
        </div>
         
        <button
          class="ui-button-modal-trigger ui-button-modal-trigger--red"
        >
          
  Contact me

        </button>
      </div>
    </div>
  </header>
   
  <div
    class="mobile-header"
  >
    <div
      class="mobile-header__top-line"
    >
      <router-link-stub
        class="mobile-header__header-logo"
        to="/"
      >
        <img
          alt="Logotype"
          src="@/assets/img/common/logo.svg"
        />
      </router-link-stub>
       
      <div
        class="mobile-header__button-wrap"
      >
        <button
          class="mobile-header__toogle-btn"
        >
          <img
            alt="Open"
            class="mobile-header__open"
            src="@/assets/img/Header/mobile-header-hamburger.svg"
          />
        </button>
      </div>
    </div>
     
    <perfect-scrollbar-stub
      class="mobile-header__scrollbar custom-scrollbar container"
    >
      <div
        class="mobile-header__content-wrap"
        style="display: none;"
      >
        <nav
          class="mobile-header__header-routes_links"
        >
          <router-link-stub
            exact=""
            to="/"
          >
            About
          </router-link-stub>
           
          <router-link-stub
            to="/services"
          >
            Services
          </router-link-stub>
           
          <router-link-stub
            to="/projects"
          >
            Projects
          </router-link-stub>
           
          <router-link-stub
            to="/careers"
          >
            Careers
          </router-link-stub>
           
          <a
            href="https://blog.maddevs.io/"
            rel="noreferrer"
            target="_blank"
          >
            Blog
          </a>
        </nav>
         
        <button
          class="ui-button-modal-trigger mobile-header__modal-trigger-btn ui-button-modal-trigger--red"
        >
          
  Contact me

        </button>
         
        <div
          class="mobile-header__contacts"
        >
          <div
            class="footer-contacts"
          >
            <div
              class="footer-contacts__call-us"
            >
              <h4
                class="footer-contacts__title"
              >
                Call us:
              </h4>
               
              <ul
                class="footer-contacts__phones-list"
              >
                <li>
                  <a
                    class="footer-contacts__phone-item footer-contacts__phone-item--flag"
                    href="tel:+1 833-MADDEVS"
                  >
                    <span
                      class="footer-contacts__icon footer-contacts__icon--usa"
                    />
                    
					+1 833-MADDEVS
				
                  </a>
                </li>
                 
                <li>
                  <a
                    class="footer-contacts__phone-item footer-contacts__phone-item--flag"
                    href="tel:+44 20 3984 8555"
                  >
                    <span
                      class="footer-contacts__icon footer-contacts__icon--uk"
                    />
                    
					+44 20 3984 8555
				
                  </a>
                </li>
              </ul>
            </div>
             
            <div
              class="footer-contacts__message-us"
            >
              <h4
                class="footer-contacts__title"
              >
                Message us:
              </h4>
               
              <ul
                class="footer-contacts__messengers-list"
              >
                <li>
                  <a
                    class="footer-contacts__messenger-item"
                    href="https://facebook.com/maddevsllc"
                    target="_blank"
                  >
                    <span
                      class="footer-contacts__message footer-contacts__message--messenger"
                    />
                    
					Messenger
				
                  </a>
                </li>
                 
                <li>
                  <a
                    class="footer-contacts__messenger-item"
                    href="http://wa.me/996555771481"
                    target="_blank"
                  >
                    <span
                      class="footer-contacts__message footer-contacts__message--whatsapp"
                    />
                    
					Whatsapp
				
                  </a>
                </li>
                 
                <li>
                  <a
                    class="footer-contacts__messenger-item"
                    href="https://t.me/mun_tamara/"
                    target="_blank"
                  >
                    <span
                      class="footer-contacts__message footer-contacts__message--telegram"
                    />
                    
					Telegram
				
                  </a>
                </li>
                 
                <li>
                  <a
                    class="footer-contacts__messenger-item"
                    href="https://msng.link/o/?https%3A%2F%2Fu.wechat.com%2FICWluRgJH8tu0IisMQ1eEFo=wc"
                    target="_blank"
                  >
                    <span
                      class="footer-contacts__message footer-contacts__message--wechat"
                    />
                    
					WeChat
				
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
         
        <div
          class="mobile-header__social-network_links"
        >
          <div
            class="footer-social-network"
          >
            <a
              class="footer-social-network__social-newtork_link"
              href="https://github.com/maddevsio"
              target="_blank"
            >
              <img
                alt="Mad Devs GitHub"
                class="footer-social-network__github"
                src="../../assets/img/Footer/github.svg"
              />
            </a>
             
            <a
              class="footer-social-network__social-newtork_link"
              href="https://www.facebook.com/maddevsllc"
              target="_blank"
            >
              <img
                alt="Mad Devs Facebook"
                class="footer-social-network__facebook"
                src="../../assets/img/Footer/facebook.svg"
              />
            </a>
             
            <a
              class="footer-social-network__social-newtork_link"
              href="https://www.instagram.com/maddevsio/"
              target="_blank"
            >
              <img
                alt="Mad Devs Instagram"
                class="footer-social-network__instagram"
                src="../../assets/img/Footer/instagram.svg"
              />
            </a>
             
            <a
              class="footer-social-network__social-newtork_link"
              href="https://blog.maddevs.io/"
              target="_blank"
            >
              <img
                alt="Mad Devs Medium"
                class="footer-social-network__medium"
                src="../../assets/img/Footer/medium.svg"
              />
            </a>
             
            <a
              class="footer-social-network__social-newtork_link"
              href="https://twitter.com/MadDevsIO"
              target="_blank"
            >
              <img
                alt="Mad Devs Twitter"
                class="footer-social-network__twitter"
                src="../../assets/img/Footer/twitter.svg"
              />
            </a>
             
            <a
              class="footer-social-network__social-newtork_link"
              href="https://www.behance.net/maddevs"
              target="_blank"
            >
              <img
                alt="Mad Devs Behance"
                class="footer-social-network__behance"
                src="../../assets/img/Footer/behance.svg"
              />
            </a>
          </div>
        </div>
      </div>
    </perfect-scrollbar-stub>
  </div>
</div>
`;
